dev/com.ibm.ws.classloading/src/com/ibm/ws/classloading/java2sec/PermissionManager.java

```diff
@@ -491,24 +485,21 @@ public String getNamespace() {
                 return Collections.emptyMap();
             }
         });
-
-        Iterator<BundleCapability> capabilitiesIterator = matchingCapabilities.iterator();
-
-        if (capabilitiesIterator.hasNext()) {
-            bundleCapability = capabilitiesIterator.next();
-        }
-
-        return bundleCapability;
+        return matchingCapabilities;
     }
 
-    private BundleWiring getBundleWiring(BundleCapability bundleCapability) {
-        BundleWiring providerBundleWiring = null;
-
-        if (bundleCapability != null) {
-            providerBundleWiring = bundleCapability.getRevision().getWiring();
+    private BundleWiring getBundleWiring(Collection<BundleCapability> bundleCapabilities) {
+        BundleCapability bundleCapability = null;
+        for (BundleCapability bc : bundleCapabilities) {
+            if (bundleCapability != null && 
+                Constants.SYSTEM_BUNDLE_LOCATION.equals(bc.getRevision().getBundle().getLocation())) {
```

Comment:
Should not use getLocation here because it requires a permission check.  Instead use `getBundleId() == 0`